Sources_v3/Controllers/MemberListController/MemberListController.swift

```diff
@@ -0,0 +1,305 @@
+            self.callback { completion?(error) }
+        }
+    }
+    
+    /// Sets the provided object as a delegate of this controller.
+    ///
+    /// - Note: If you don't use custom extra data types, you can set the delegate directly using `controller.delegate = self`.
+    /// Due to the current limits of Swift and the way it handles protocols with associated types, it's required to use this
+    /// method to set the delegate, if you're using custom extra data types.
+    ///
+    /// - Parameter delegate: The object used as a delegate. It's referenced weakly, so you need to keep the object
+    /// alive if you want keep receiving updates.
+    ///
+    public func setDelegate<Delegate: _ChatChannelMemberListControllerDelegate>(_ delegate: Delegate)
+        where Delegate.ExtraData == ExtraData {
+        multicastDelegate.mainDelegate = AnyChatChannelMemberListControllerDelegate(delegate)
+    }
+    
+    private func createMemberListUpdater() -> ChannelMemberListUpdater<ExtraData> {
+        environment.memberListUpdaterBuilder(
+            client.databaseContainer,
+            client.webSocketClient,
+            client.apiClient
+        )
+    }
+    
+    private func createMemberListObserver() -> ListDatabaseObserver<_ChatChannelMember<ExtraData.User>, MemberDTO> {
+        let observer = environment.memberListObserverBuilder(
+            client.databaseContainer.viewContext,
+            MemberDTO.members(matching: query),
+            { $0.asModel() },
+            NSFetchedResultsController<MemberDTO>.self
+        )
+        
+        observer.onChange = { [unowned self] changes in
+            self.delegateCallback {
+                $0.memberListController(self, didChangeMembers: changes)
+            }
+        }
+        
+        return observer
+    }
+    
+    private func startObservingIfNeeded() {
+        guard state == .initialized else { return }
+        
+        do {
+            try memberListObserver.startObserving()
+            state = .localDataFetched
+        } catch {
+            log.error("Observing members matching <\(query)> failed: \(error). Accessing `members` will always return `[]`.")
+            state = .localDataFetchFailed(ClientError(with: error))
+        }
+    }
+}
+
+// MARK: - Actions
+
+public extension _ChatChannelMemberListController {
+    /// Loads next members from backend.
+    /// - Parameters:
+    ///   - limit: The page size.
+    ///   - completion: The completion. Will be called on a **callbackQueue** when the network request is finished.
+    ///                 If request fails, the completion will be called with an error.
+    func loadNextMembers(
+        limit: Int = 25,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        var updatedQuery = query
+        updatedQuery.pagination = [.limit(limit), .offset(members.count)]
+        memberListUpdater.load(updatedQuery) { [weak self] error in
+            self?.query = updatedQuery
```

Comment:
We don't update queries in other `***ListController` types. Seems like we do need to update to keep it relevant.

Reply:
Hmmm, why do you think we need to update it? The only thing that changes is the pagination options and we don't need to persist it.

Reply:
Decided to change the `controller.query` because `Pagination.limit/offset` are public and I assumed that end users might use these field -> would be nice to have them up-to-date